ctapipe/irf/irf_classes.py

```diff
@@ -0,0 +1,389 @@
+            ("valid geom reco", "HillasReconstructor_is_valid"),
+            ("valid energy reco", "RandomForestRegressor_is_valid"),
+        ],
+        help=QualityQuery.quality_criteria.help,
+    ).tag(config=True)
+
+    rename_columns = List(
+        help="List containing translation pairs new and old column names"
+        "used when processing input with names differing from the CTA prod5b format"
+        "Ex: [('valid_geom','HillasReconstructor_is_valid')]",
+        default_value=[],
+    ).tag(config=True)
+
+    def normalise_column_names(self, events):
+        keep_columns = [
+            "obs_id",
+            "event_id",
+            "true_energy",
+            "true_az",
+            "true_alt",
+        ]
+        rename_from = [
+            f"{self.energy_reconstructor}_energy",
+            f"{self.geometry_reconstructor}_az",
+            f"{self.geometry_reconstructor}_alt",
+            f"{self.gammaness_classifier}_prediction",
+        ]
+        rename_to = ["reco_energy", "reco_az", "reco_alt", "gh_score"]
+
+        # We never enter the loop if rename_columns is empty
+        for new, old in self.rename_columns:
+            rename_from.append(old)
+            rename_to.append(new)
+
+        keep_columns.extend(rename_from)
+        events = QTable(events[keep_columns], copy=False)
+        events.rename_columns(rename_from, rename_to)
+        return events
+
+    def make_empty_table(self):
+        """This function defines the columns later functions expect to be present in the event table"""
+        columns = [
+            "obs_id",
+            "event_id",
+            "true_energy",
+            "true_az",
+            "true_alt",
+            "reco_energy",
+            "reco_az",
+            "reco_alt",
+            "gh_score",
+            "pointing_az",
+            "pointing_alt",
+            "theta",
+            "true_source_fov_offset",
+            "reco_source_fov_offset",
+            "weight",
+        ]
+        units = [
+            None,
+            None,
+            u.TeV,
+            u.deg,
+            u.deg,
+            u.TeV,
+            u.deg,
+            u.deg,
+            None,
+            u.deg,
+            u.deg,
+            u.deg,
+            u.deg,
+            u.deg,
+            None,
+        ]
+
+        return QTable(names=columns, units=units)
+
+
+class ThetaSettings(Component):
+
+    min_angle = Float(
+        default_value=0.05, help="Smallest angular cut value allowed"
+    ).tag(config=True)
+    max_angle = Float(default_value=0.32, help="Largest angular cut value allowed").tag(
+        config=True
+    )
+    min_counts = Integer(
+        default_value=10,
+        help="Minimum number of events in a bin to attempt to find a cut value",
+    ).tag(config=True)
+    fill_value = Float(
+        default_value=0.32, help="Angular cut value used for bins with too few events"
+    ).tag(config=True)
+
+
+class DataBinning(Component):
```

Comment:
I'd rather split this into classes per IRF. We also will need to support different parametrizations of the different IRFs, which will need their own configurations.

Reply:
I just split out the energy binning for now, as I wasn't sure how to group the rest.